### PR TITLE
Basic redux - for movies

### PR DESCRIPTION
**Pull request by maxnatalia**
I added basic redux and installed axios, I implemented moviesSlice and moviesSaga - this is my suggestion, how to create this kind of components.

### CODE REVIEW

**Review by kraviecdev — APPROVED** [review]
installed axios + API url i API key, all good

File: src/features/movies/moviesSaga.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import axios from "axios";
+import { takeLatest, put } from "redux-saga/effects";
+import { fetchError, fetchMoviesSuccess, fetchMovieSuccess } from "./moviesSlice";
+import { apiUrl, apiKey } from "../../apiData";
+
+function* fetchMoviesHandler() {
+    try {
+        const data = yield axios.get(
+            `${apiUrl}movie/popular?api_key=${apiKey}&language=en-US&page=1`
```
Comment (kraviecdev) [inline src/features/movies/moviesSaga.js:9]: for this, i would rather make a fn asyn/await in other js file for api and then use it in saga

File: src/features/movies/moviesSlice.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+const moviesSlice = createSlice({
+    name: "movies",
+    initialState: {
+        loading: true,
+        error: false,
+        movies: [],
```
Comment (kraviecdev) [inline src/features/movies/moviesSlice.js:8]: we need also initial state for single movie and reducer for it

File: src/index.js [hunk 3]
```diff
@@ -2,11 +2,15 @@ import React from 'react';
...
 import App from './App';
 import reportWebVitals from './reportWebVitals';
+import { Provider } from 'react-redux';
+import store from './store';
 
 const root = ReactDOM.createRoot(document.getElementById('root'));
 root.render(
   <React.StrictMode>
-    <App />
+    <Provider store={store}>
```
Comment (kraviecdev) [inline src/index.js:11]: good implementation

File: src/features/movies/moviesSaga.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        const movie = yield axios.get(
+            `${apiUrl}movie/${id}?api_key=${apiKey}&language=en-US`
+        );
+        yield put(fetchMovieSuccess({ payload: movie }));
+    } catch (error) {
+        yield put(fetchError());
+    }
+}
+
+export function* watchFetchPopularMovies() {
```
Comment (kraviecdev) [inline src/features/movies/moviesSaga.js:33]: there is no import fetchMovies & and fetchMovie from slice, 
Exported fn should have same name as in rootSaga - moviesSaga()

**Review by kraviecdev — CHANGES_REQUESTED** [review]
comments in code = changes requaried

File: src/features/movies/moviesSaga.js (hunk 1 above)
Comment (maxnatalia) [reply src/features/movies/moviesSaga.js:9]: ok, we can do this with async/await and add other file

File: src/features/movies/moviesSlice.js (hunk 2 above)
Comment (maxnatalia) [reply src/features/movies/moviesSlice.js:8]: we have initialstate for movie here

File: src/features/movies/moviesSaga.js (hunk 4 above)
Comment (maxnatalia) [reply src/features/movies/moviesSaga.js:33]: this is only example implementation I added comments in root

**Review by kraviecdev — APPROVED** [review]
Merge basic redux & saga

**Pull request merged**